src/encoding.ts

```diff
@@ -98,15 +98,15 @@ export interface Encoding extends UnitEncoding {
 export function has(encoding: Encoding, channel: Channel): boolean {
   const channelEncoding = encoding && encoding[channel];
   return channelEncoding && (
-    channelEncoding.field !== undefined ||
+    'field' in channelEncoding ||
     // TODO: check that we have field in the array
     (isArray(channelEncoding) && channelEncoding.length > 0)
   );
 }
 
 export function isAggregate(encoding: Encoding) {
   return some(CHANNELS, (channel) => {
-    if (has(encoding, channel) && encoding[channel].aggregate) {
+    if (has(encoding, channel) && 'aggregate' in encoding[channel]) {
```

Comment:
Error TS2339: Property 'aggregate' does not exist on type 'FieldDef | PositionChannelDef | ChannelDefWithLegend | OrderChannelDef | FieldDef[] | OrderChanne...'.
  Property 'aggregate' does not exist on type 'FieldDef[]'.